File: .NET-app/MorkosiaPrepaLeague/Models/Core/TransferWindow.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace MorkosiaPrepaLeague.Models.Core
{
    public class TransferWindow : CosmosDocumentBase
    {
        [Required]
        public string LeagueId { get; set; } = string.Empty;
        
        /// <summary>
        /// Start date and time of the transfer window in UTC
        /// </summary>
        [Required]
        public DateTime StartDate { get; set; }
        
        /// <summary>
        /// End date and time of the transfer window in UTC
        /// </summary>
        [Required]
        public DateTime EndDate { get; set; }
        
        public int WindowNumber { get; set; }
       
        public string CreatedByAdminId { get; set; } = string.Empty;
    }
}

File: src/app/models/transfer_window.py
from datetime import datetime

from beanie import Indexed
from pydantic import Field

from app.models.base import DocumentBase


class TransferWindow(DocumentBase):
    league_id: Indexed(str) = ""  # type: ignore[valid-type]
    start_date: datetime
    end_date: datetime
    window_number: int = 0
    created_by_admin_id: str = ""

    class Settings:
        name = "transfer_windows"
